### PR TITLE
Datatable & SendBetterEmail PMD fixes

### PR DESCRIPTION
**Pull request by dschach**
Some standard rules for PMD are now fixed - easier to use this unlocked package in CI tools

### CODE REVIEW

**Comment by ericrsmith35** [comment]
I'm fine with the Datatable syntax changes but I can't comment on the Send Better Email changes.

**Comment by dschach** [comment]
@ericrsmith35 @jdpond Requesting approvals for this. I can't specify approver names here.

**Review by alexed1 — COMMENTED** [review]
Hi @dschach! thank you for this PR!


I like the idea of making this code easier to use with CI systems, but want to better understand the changes that are being made here.

I'm not familiar with PMD....I see it's a static analysis tool. I can't tell what real changes, if any, are proposed here because of the tab insertion.

it looks like your PR inserts a lot of tabs, pushing code to the right. I think that I'd like to avoid that.

Can you help me come up to speed on this? many thanks....

**Comment by dschach** [comment]
@alexed1 PMD is generally considered the standard for static code analysis. It finds common problems in code, such as if statements that don't use brackets. One of those mistakes brought down all of iCloud a few years ago. You can find more info at [https://pmd.github.io/](https://pmd.github.io/)
Looking at the changes, I don't see much code moving right. It's just converting whitespace from spaces to tabs, which is a more common practice in some circles. 
I added code comments in keeping with ApexDoc/JavaDoc principles.
I think you'll find the code is a bit easier to read and is closer to meeting Salesforce's recommended naming conventions (which generally mimic Java's conventions).

I love the code in UnofficialSF, but when we install unlocked packages, because the code is visible, if it does not adhere to PMD standards (because many customers use PMD scanning to ensure proper code quality) it can make CI tools fail. That's why I'm hoping to help with refactoring code as we use it in our org and have to put in brackets or remove unused variables, etc. 

You can find the PMD rules [on their website](https://pmd.github.io/latest/pmd_rules_apex.html) and I'd be happy to set up a call to show you the tool, its uses, and how many Salesforce customers use it in our development processes.

**Comment by alexed1** [comment]
I'm definitely open to the idea....it has always been a point of frustration to me that many customers can't use the unmanaged code, while putting it on app exchange is very labor intensitve. 

looking at the rules I don't see anything about Tabs vs Spaces. Can you redo the PR without the Tabs change? I see that having two benefits:
1) it holds off on a change to spacing vs. tabs that would fragment the style of the repo (i.e. if we're going to change to tabs, we should do it for all the code at once potentially and spread the word about the new style)
2) I don't have a lot of time to read code, and without all of the tabs/spaces changes, I'll be able to more effectively see the meaningful changes that are being proposed here.

**Comment by jkranz-rk** [comment]
Agree with @alexed1 on this. PMD compliance is fantastic and I fully support it, but tabs vs spaces is not a PMD rule that I'm familiar with. Perhaps Prettier was applied? 

For what its worth on this age-old tabs vs spaces debate: I tend to prefer tabs but I'm not sure how much that is the norm in the SF ecosystem and can deal with either.  As long as its consistent and easy to comply with by providing a sensible prettier config file. 

(side note @alexed1: one of the annoying default settings on GitHub is to display tabs as 8 spaces. If you haven't adjusted that, try setting that to display tabs as 4 spaces, it makes things much more readable. Settings -> Appearance -> Tab Size Preference.)

**Comment by dschach** [comment]
@jkranz-rk That's great feedback. Yeah, I sometimes forget that I run Prettier by default. I will move it back to spaces for now, but I do think that it would be a good exercise to generate code standards for this repo, as it is used by so many people. I'd be happy to generate a Prettier config file (using spaces at first, though I'm a tabs person through and through) in a future PR.

**Comment by dschach** [comment]
I have updated to use spaces.
@alexed1 There is a feature in the comparison tab where you can hide whitespace changes if you'd like - it lets you focus on the code itself.

Would anyone be up for collaborating on a code standards doc for this repo? It would be helpful as it grows. It could include formatting, PMD rules that should not be violated, best-practices (e.g. brackets around every if block), and more.

**Comment by ericrsmith35** [comment]
I don't currently have time for the collaboration but I'd be glad to see and try and follow the results.

**Comment by alexed1** [comment]
> @alexed1 There is a feature in the comparison tab where you can hide whitespace changes if you'd like - it lets you focus on the code itself.

VERY useful tip!

**Comment by alexed1** [comment]
I like the idea of a set of standards. What are the practical implications, once a set of standards is established. do we tell contributors to make sure to add a specific prettier file to their default sfdx project? can that be done automatically? It then makes changes at save time?

File: flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls [hunk 1]
```diff
@@ -1,194 +1,181 @@
...
- * 2.00.02	8/15/2020	Jack Pond, Mohith		Modified for coverage testing
- * 2.00.02	8/16/2020	Jack Pond				Corrected singleEmail recipient name for templateTargetId
- * 2.00.02	10-06-2020	Jack Pond				Reverted naming, fixed bugs
- * 2.00.03  11-28-2020  Jack Pond               Updated for Flow Action BasePack and Flow Screen Component Base Pack.
- * 2.00.05  11-28-2020  Jack Pond               Added setTreatTargetObjectAsRecipient Fix: #586
- * 2.00.05  02-15-2021  Jack Pond               ReplyEmail with SendBetterEmail #595
- * 2.01.02  2021-06-11  Jack Pond               Modified so that all checks for null also check for .length() > 0 #702
- * 2.01.05  2021-06-29  Jack Pond               Pasted Typo fix #797 for null pointer exception
- * 2.01.06	2021-07-30	Jack D. Pond		    Fix remainder of checkboxes default/save Fix: #702,#831,#785,#829
+ * @since 0.01					Alex Edelstein			Original Author
```
Comment (alexed1) [inline flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls:11]: What does @since do?

**Review by alexed1 — APPROVED** [review]

**Pull request merged**